src/main/flight/pid.c

```diff
@@ -207,55 +210,67 @@ void pidInitFilters(const pidProfile_t *pidProfile)
 
     if (dTermNotchHz != 0 && pidProfile->dterm_notch_cutoff != 0) {
         static biquadFilter_t biquadFilterNotch[2];
-        dtermNotchFilterApplyFn = (filterApplyFnPtr)biquadFilterApply;
+        dtermNotchApplyFn = (filterApplyFnPtr)biquadFilterApply;
         const float notchQ = filterGetNotchQ(dTermNotchHz, pidProfile->dterm_notch_cutoff);
         for (int axis = FD_ROLL; axis <= FD_PITCH; axis++) {
-            dtermFilterNotch[axis] = &biquadFilterNotch[axis];
-            biquadFilterInit(dtermFilterNotch[axis], dTermNotchHz, targetPidLooptime, notchQ, FILTER_NOTCH);
+            dtermNotch[axis] = &biquadFilterNotch[axis];
+            biquadFilterInit(dtermNotch[axis], dTermNotchHz, targetPidLooptime, notchQ, FILTER_NOTCH);
         }
     } else {
-        dtermNotchFilterApplyFn = nullFilterApply;
+        dtermNotchApplyFn = nullFilterApply;
     }
-
-    static dtermFilterLpf_t dtermFilterLpfUnion;
-    if (pidProfile->dterm_lpf_hz == 0 || pidProfile->dterm_lpf_hz > pidFrequencyNyquist) {
-        dtermLpfApplyFn = nullFilterApply;
+    
+    //2nd Dterm Lowpass Filter
+	static pt1Filter_t dtermLowpass2State;
+	if (pidProfile->dterm_lowpass2_hz == 0 || pidProfile->dterm_lowpass2_hz > pidFrequencyNyquist) {
+		dtermLowpass2ApplyFn = nullFilterApply;
+	} else {
+		dtermLowpass2ApplyFn = (filterApplyFnPtr)pt1FilterApply;
+		for (int axis = FD_ROLL; axis <= FD_PITCH; axis++) {
+			dtermLowpass2[axis] = &dtermLowpass2State;
```

Comment:
This falls short of using the same state for all axes, just like @fujin's PR before